chart/templates/_commonChecks.tpl

```diff
@@ -510,28 +526,72 @@ Return customer values to use in preflights and support-bundle
 Return customer secrets to use in preflights and support-bundle
 */}}
 {{- define "carto.replicated.tenantRequirementsChecker.customerSecrets" }}
+  {{- if eq .Values.externalPostgresql.existingSecret "" -}}
   - name: WORKSPACE_POSTGRES_PASSWORD
     value: {{ .Values.externalPostgresql.password | quote }}
+  {{- else }}
+  - name: WORKSPACE_POSTGRES_PASSWORD
+    valueFrom:
+      secretKeyRef:
+        name: {{ include "carto.postgresql.secretName" . }}
+        key: {{ include "carto.postgresql.secret.key" . }}
+  {{- end -}}
+  {{- if eq .Values.externalRedis.existingSecret "" -}}
   - name: REDIS_PASSWORD
     value: {{ .Values.externalRedis.password | quote }}
+  {{- else }}
+  - name: REDIS_PASSWORD
+    valueFrom:
+      secretKeyRef:
+        name: {{ include "carto.redis.secretName" . }}
+        key: {{ include "carto.redis.existingsecret.key" . | quote }}
+  {{- end -}}
+  {{- if eq .Values.cartoSecrets.launchDarklySdkKey.existingSecret.name "" }}
   - name: LAUNCHDARKLY_SDK_KEY
     value: {{ .Values.cartoSecrets.launchDarklySdkKey.value | quote }}
-  {{- if eq .Values.appConfigValues.storageProvider "s3" }}
+  {{- else -}}
+  {{ include "carto._utils.generateSecretDef" (dict "var" "LAUNCHDARKLY_SDK_KEY" "context" .) | nindent 2 }}
+  {{- end -}}
+  {{- if eq .Values.appConfigValues.storageProvider "s3" -}}
+  {{- if eq .Values.appSecrets.awsAccessKeyId.existingSecret.name "" }}
```

Comment:
A trick for the future to improbe readability, you can add additional whitespaces to properly read it. Example:
```
{{- if eq .Values.appConfigValues.storageProvider "s3" -}}
{{-   if eq .Values.appSecrets.awsAccessKeyId.existingSecret.name "" }}
      - xyz
{{-   end -}}
{{-   if eq .Values.appSecrets.awsAccessKeyId.existingSecret.name "" }}
      - xyz
{{-   end -}}
{{- end -}}
```

Reply:
If I add these additional whitespaces then the template is generated with them 😓 I'm not sure if there is any way we can do it 🤔